common/openapi.yml

```diff
@@ -0,0 +1,51 @@
+openapi: 3.0.3
+info:
+  title: Terra Batch Analysis
+  description: Manage batch analyses, workflows, and submissions in userspace
+  version: 0.0.1
+paths:
+  /api/batch/v1/methods:
+    get:
+      summary: Lists known methods
+      tags: [ batch ]
+      operationId: listMethods
+      responses:
+        '200':
+          description: A JSON array of methods
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+components:
+  responses:
+
+    NotFound:
+      description: Not found (or unauthorized)
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorReport'
+    ServerError:
+      description: Server error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorReport'
+
+  schemas:
+
+    ErrorReport:
```

Comment:
This (and catalog) follows the ECM's `ErrorReport` model. Other projects like BPM have a `causes` field that contains an array of all the exceptions up the chain. The TCL's global exception handler supports gathering up these exceptions so I've been thinking about changing catalog to use the BPM model instead.

Reply:
What is BPM?